src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -520,7 +542,12 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
         newDevice.stream = stream;
         newDevice.constraints = proposedConstraints;
       } else {
-        newDevice.stream = await navigator.mediaDevices.getUserMedia(proposedConstraints);
+        if (kind === 'audio' && this.activeDevices[kind]) {
```

Comment:
Why do we need to do this here? This should be done in releaseMediaStream already

Reply:
We need to stop the active MediaTrack at least once before we call the getUserMedia(constraint)
This call tells the user agent that the track's source is no longer needed by the MediaStreamTrack

Reply:
- A good point - SDK is already releasing an old audio stream at line 618.
- Assume you are using Microphone1, and the browser doesn't store a previously-selected permisison, e.g. FF. Now, you are choosing Microphone2. If you release an active input here, you might be muted until you allow the microphone permission opened from `getUserMedia`.

Reply:
For using 'default', once the MediaStreamTrack reference is removed, only then the mapping between the ‘default’ and the original device is dissociated from the browser.
I can add another check like:
JSON.parse(JSON.stringify(proposedConstraints.audio)).deviceId === 'default' before the stop is called

Reply:
Could you elaborate this a bit more? Is this for fixing the Chrome issue where calling `getUserMedia` with `default` doesn't work when another `default` is already selected?